core/embed/trezorhal/stm32f4/flash.c

```diff
@@ -182,6 +305,37 @@ secbool flash_write_word(uint8_t sector, uint32_t offset, uint32_t data) {
   return sectrue;
 }
 
+secbool flash_area_write_byte(const flash_area_t *area, uint32_t offset,
+                              uint8_t data) {
+  uint16_t sector;
+  uint32_t sector_offset;
+  if (get_sector_and_offset(area, offset, &sector, &sector_offset) != sectrue) {
+    return secfalse;
+  }
+  return flash_write_byte(sector, sector_offset, data);
+}
+
+secbool flash_area_write_word(const flash_area_t *area, uint32_t offset,
+                              uint32_t data) {
+  uint16_t sector;
+  uint32_t sector_offset;
+  if (get_sector_and_offset(area, offset, &sector, &sector_offset) != sectrue) {
```

Comment:
I don't want to get into premature optimization, but I feel I should mention that when writing a whole sector, e.g. during firmware installation, `get_sector_and_offset()` will get called 32768 times, even though it needs to be called only once. I recommend to measure the firmware installation time before and after this PR. If it introduces a noticeable slowdown, then I recommend defining a function which writes a sequence of words, like
```c
secbool flash_area_write_words(const flash_area_t *area, uint32_t offset,
                               uint32_t *words, size_t word_count);
```
We can debate whether it should be able to traverse sectors. FTR, if we get into optimization, then we can also get rid of multiple calls to `flash_get_address()` and alignment checks.

Reply:
That's about a 5 % increase in time. I agree it's negligible.